### DISCUSSION TITLE
Can't seem to build latest code

### DISCUSSION THREAD

**Discussion by meetsubhojit** [apollographql/router#2093] [Q&A] [answered]
Can't seem to build the latest clone code using `cargo build`. Always getting below error in opentelemetry. 

```
error: failed to run custom build command for `opentelemetry-otlp v0.10.0`

Caused by:
  process didn't exit successfully: `./apollographql/router-copy/router/target/debug/build/opentelemetry-otlp-673d1aad0ede2b10/build-script-build` (exit status: 1)
  --- stderr
  error running rustfmt: Os { code: 2, kind: NotFound, message: "No such file or directory" }
warning: build failed, waiting for other jobs to finish...
error: failed to run custom build command for `apollo-router v1.3.0 (./apollographql/router-copy/router/apollo-router)`

Caused by:
  process didn't exit successfully: `./apollographql/router-copy/router/target/debug/build/apollo-router-2eed5c48e4f7d632/build-script-main` (exit status: 1)
  --- stdout
  cargo:rerun-if-changed=./apollographql/router-copy/router/apollo-router/src/spaceport/proto/agents.proto
  cargo:rerun-if-changed=.apollographql/router-copy/router/apollo-router/src/spaceport/proto/reports.proto

  --- stderr
  error running rustfmt: Os { code: 2, kind: NotFound, message: "No such file or directory" }
```

any idea?

**Comment by abernix** [comment, marked as answer]
It appears that something in OpenTelemetry's build dependencies is calling either `cargo fmt` or `rustfmt` (directly).  The command seems to indicate that `rustfmt` doesn't exist on your machine.  I would love to have a better error message around this or documentation, but I'm not sure where it's coming from precisely.

To unblock you though, does running the following command — and then trying a new build — fix this for you?

```
rustup component add rustfmt
```

**Reply by meetsubhojit** [reply]
@abernix thnx for pointing out the issue. after knowing the issue the debugging was much easier.
my problem seems to be that i installed rust using `brew` and it didn't seem to install components like rustup, rustfmt etc. I had to uninstall via brew and installed rust again via `rustup.rs`. Rebuilding the code works fine now. thanks.
